src/downloader/message_decoder.rs

```diff
@@ -63,4 +88,284 @@ mod tests {
+    //     let expected_bytes =
+    //         hex!("f847820457f842a000000000000000000000000000000000000000000000000000000000deadc0dea000000000000000000000000000000000000000000000000000000000feedbeef");
+    //     let result = decode_rlp_message(EthMessageId::GetNodeData, &expected_bytes);
+    //     let some_message = result.unwrap();
+    //     let bytes = rlp::encode(&some_message);
+    //     assert_eq!(&*bytes, expected_bytes);
+
+    //     assert_eq!(
+    //         some_message,
+    //         Message::GetNodeData(GetNodeDataMessage {
+    //             request_id: 1111,
+    //             hashes: vec![
+    //                 H256(hex!(
+    //                     "00000000000000000000000000000000000000000000000000000000deadc0de"
+    //                 )),
+    //                 H256(hex!(
+    //                     "00000000000000000000000000000000000000000000000000000000feedbeef"
+    //                 ))
+    //             ]
+    //         })
+    //     );
+    // }
+
+    #[test]
+    fn decode_get_receipts() {
+        let expected_bytes =
+        hex!("f847820457f842a000000000000000000000000000000000000000000000000000000000deadc0dea000000000000000000000000000000000000000000000000000000000feedbeef");
+        let result = decode_rlp_message(EthMessageId::GetReceipts, &expected_bytes);
+        let some_message = result.unwrap();
+        let bytes = rlp::encode(&some_message);
+        assert_eq!(&*bytes, expected_bytes);
+
+        assert_eq!(
+            some_message,
+            Message::GetReceipts(GetReceiptsMessage {
+                request_id: 1111,
+                block_hashes: vec![
+                    H256(hex!(
+                        "00000000000000000000000000000000000000000000000000000000deadc0de"
+                    )),
+                    H256(hex!(
+                        "00000000000000000000000000000000000000000000000000000000feedbeef"
+                    ))
+                ]
+            })
+        );
+    }
+
+    #[test]
+    fn decode_receipts() {
+        let expected_bytes =
+        hex!("f90172820457f9016cf90169f901668001b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000f85ff85d940000000000000000000000000000000000000011f842a0000000000000000000000000000000000000000000000000000000000000deada0000000000000000000000000000000000000000000000000000000000000beef830100ff");
+        let result = decode_rlp_message(EthMessageId::Receipts, &expected_bytes);
+        let some_message = result.unwrap();
```

Comment:
This decode operation fails for reasons that aren't entirely clear to me. The observed error type is `RlpExpectedToBeList`.  I suspect that the problem may be in the RlpDerive macro, since `decode_node_data` also fails and both of those types have entries of the form `Vec<Vec<T>>`, but I don't have any solid leads.

Reply:
Update: the problem has to do with the definition of `Receipt` in the [ethereum crate](https://docs.rs/ethereum/0.9.0/ethereum/struct.Receipt.html). An RLP encoded receipt from the crate represents the `state_root` as a 32 byte list, while the EIP 2481 test case represents it as a single zero byte. 

I believe the fix is to change the definition of `ethereum::Receipt` to:
```
pub struct Receipt {
	pub state_root: U256, // This used to be H256
	pub used_gas: U256,
	pub logs_bloom: Bloom,
	pub logs: Vec<Log>,
}
```

Reply:
Confirmed: Making this change locally causes the test to pass. I've opened an [issue](https://github.com/rust-blockchain/ethereum/issues/19) on the `ethereum` repo and will submit a PR to fix shortly.

Reply:
:+1: